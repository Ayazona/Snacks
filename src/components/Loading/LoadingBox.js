import Radium from 'radium'
import { colors } from 'styles'
import PropTypes from 'prop-types'
import React, { PureComponent } from 'react'

const buildKeyFramesStyles = (startColor, endColor, name) => ({
  animationName: Radium.keyframes({
    '0%': { backgroundColor: startColor },
    '50%': { backgroundColor: endColor },
    '100%': { backgroundColor: startColor },
  }, name),
  backgroundColor: startColor,
})

const baseLoadingStyle = {
  animation: 'x 3s ease-in-out infinite',
  ...buildKeyFramesStyles(colors.GRAY_93, colors.GRAY_88, 'loading'),
  width: '100%',
  height: '20px'
}

const darkStyles = buildKeyFramesStyles(colors.GRAY_88, colors.GRAY_74, 'darkLoading')
const lightStyles = buildKeyFramesStyles(colors.GRAY_97, colors.GRAY_93, 'lightLoading')

@Radium
class LoadingBox extends PureComponent {
  static propTypes = {
    /** Use for rendering dark backgrounds. */
    dark: PropTypes.bool,

    /** Use for rendering light backgrounds, overrides dark */
    light: PropTypes.bool,

    /** Optional style overrides. */
    style: PropTypes.object,
  }

  static defaultProps = {
    dark: false,
    light: false,
  }

  componentWillReceiveProps({ dark, light }) {
    if (dark && light) {
      throw new TypeError('LoadingBox can not receive both dark and light props')
    }
  }

  shouldComponentUpdate({ dark, light, style }) {
    const { dark: oldDark, light: oldLight, style: oldStyle } = this.props
    return dark !== oldDark || light !== oldLight || style !== oldStyle
  }

  render() {
    const { dark, light, style } = this.props

    const boxStyle = {
      ...baseLoadingStyle,
      ...(dark ? darkStyles : {}),
      ...(light ? lightStyles : {}),
      ...style,
    }

    return (
      <div style={boxStyle} />
    )
  }
}

export default LoadingBox
