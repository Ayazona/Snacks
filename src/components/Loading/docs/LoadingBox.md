### LoadingBox example:
Light:
    <div>
      <LoadingBox light />
    </div>

Standard:
    <div>
      <LoadingBox />
    </div>

Dark:
    <div>
      <LoadingBox dark />
    </div>

Example Combination:

    const cardStyles = {
      display: 'flex',
      flexDirection: 'row',
      width: '330px'
    };
    const baseLineStyle = { width: 200, marginTop: 10 };
    const LoadingLine = () => <LoadingBox style={baseLineStyle} />;

    <div style={cardStyles}>
      <div style={{ flexGrow: 1 }}>
        <LoadingBox style={{ width: '100px' }} dark />
        <LoadingLine />
        <LoadingLine />
        <LoadingLine />
      </div>
      <div>
        <LoadingBox style={{ borderRadius: '50%', height: 100, width: 100, marginBottom: 10 }} light />
      </div>
    </div>
