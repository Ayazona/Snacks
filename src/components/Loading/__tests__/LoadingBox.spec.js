import React from 'react'
import renderer from 'react-test-renderer'
import LoadingBox from '../LoadingBox'
import { StyleRoot } from 'radium'

it('renders the standard LoadingBox correctly', () => {
  const tree = renderer.create(
    <StyleRoot>
      <LoadingBox style={{ width: '50px' }} />
    </StyleRoot>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders the standard LoadingBox light', () => {
  const tree = renderer.create(
    <StyleRoot>
      <LoadingBox light style={{ width: '50px' }} />
    </StyleRoot>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders the standard LoadingBox dark', () => {
  const tree = renderer.create(
    <StyleRoot>
      <LoadingBox dark style={{ width: '50px' }} />
    </StyleRoot>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
