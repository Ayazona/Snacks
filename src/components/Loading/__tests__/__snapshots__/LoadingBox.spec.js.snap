// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the standard LoadingBox correctly 1`] = `
<div
  data-radium={true}
>
  <div
    data-radium={true}
    style={
      Object {
        "animation": "x 3s ease-in-out infinite",
        "animationName": "loading-radium-animation-2dabe53b",
        "backgroundColor": "#EEEEEE",
        "height": "20px",
        "width": "50px",
      }
    }
  />
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "@keyframes loading-radium-animation-2dabe53b {
      0%{background-color: #EEEEEE;}
      50%{background-color: #E0E0E0;}
      100%{background-color: #EEEEEE;}
      }
      ",
      }
    }
  />
</div>
`;

exports[`renders the standard LoadingBox dark 1`] = `
<div
  data-radium={true}
>
  <div
    data-radium={true}
    style={
      Object {
        "animation": "x 3s ease-in-out infinite",
        "animationName": "darkLoading-radium-animation-f5235588",
        "backgroundColor": "#E0E0E0",
        "height": "20px",
        "width": "50px",
      }
    }
  />
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "@keyframes darkLoading-radium-animation-f5235588 {
      0%{background-color: #E0E0E0;}
      50%{background-color: #BDBDBD;}
      100%{background-color: #E0E0E0;}
      }
      ",
      }
    }
  />
</div>
`;

exports[`renders the standard LoadingBox light 1`] = `
<div
  data-radium={true}
>
  <div
    data-radium={true}
    style={
      Object {
        "animation": "x 3s ease-in-out infinite",
        "animationName": "lightLoading-radium-animation-438c49ee",
        "backgroundColor": "#F7F7F7",
        "height": "20px",
        "width": "50px",
      }
    }
  />
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "@keyframes lightLoading-radium-animation-438c49ee {
      0%{background-color: #F7F7F7;}
      50%{background-color: #EEEEEE;}
      100%{background-color: #F7F7F7;}
      }
      ",
      }
    }
  />
</div>
`;
